### PR TITLE
Added Painting Fence Code for C++/CPP

### LINKED ISSUE #3397: Painting Fence Algorithm in  C++/CPP
## 💥 Proposal

I would like to implement Painting Fence Algorithm in  C++/CPP as a part of Gssoc21.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by nandakishormpai**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added the code for Painting Fence Algorithm in C++/CPP

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #3397

### CODE REVIEW

**Comment by nandakishormpai** [comment]
@HarshCasper @ankitaggarwal23 @bhav09  Kindly review my pull request.

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Output:
+result: number of ways of painting
+        considering the condition above
+
+*/
+
+#include <iostream>
+using namespace std;
+
+int paint_fence(int n, int k){
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp]: Just add one line description of your function

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    int result = paint_fence(post,color);
+    if(result==-1){
+        cout<<"Painting is not possible following the condition\n";
+    }
+    else{
+        cout<<"Number of Possiblities: "<<result<<"\n";
+    }
+
+    return 0;
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp]: Remove one extra line of space

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Painting is not possible following the condition
+
+Enter n,Number of Posts
+5
+Enter k,Number of Colors
+3
+Number of Possiblities: 180
+
+Time Complexity  O(n)
+Space Complexity O(n) */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp]: Add a new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by nandakishormpai** [comment]
@ankitaggarwal23  Thanks for your review. I have made necessary changes.

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+Output:
+result: number of ways of painting
+        considering the condition above
+
+*/
+
+#include <iostream>
+using namespace std;
+
+int paint_fence(int n, int k){          //function uses permutation logic to output number of possible ways of painting
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp]: Shift the comment above the function and after shifting just add a new line after `using namespace std;`

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        cout<<"Painting is not possible following the condition\n";
+    }
+    else{
+        cout<<"Number of Possiblities: "<<result<<"\n";
+    }
+    return 0;
+}
+
+
+/*Sample Input/Output
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp:57]: Shift `Sample Input/Output` to new line

File: C-Plus-Plus/cp/Paint_Fence.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+Painting is not possible following the condition
+
+Enter n,Number of Posts
+5
+Enter k,Number of Colors
+3
+Number of Possiblities: 180
+
+Time Complexity  O(n)
+Space Complexity O(n) */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Paint_Fence.cpp]: Shift `*/` to new line

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge

File: C-Plus-Plus/cp/Paint_Fence.cpp (hunk 6 above)
Comment (nandakishormpai) [reply C-Plus-Plus/cp/Paint_Fence.cpp]: okay sure.

**Comment by nandakishormpai** [comment]
@ankitaggarwal23  thank you for your timely response. I have made necessary changes

**Comment by ankitaggarwal23** [comment]
> @ankitaggarwal23 thank you for your timely response. I have made necessary changes

@nandakishormpai2001 Change the above two requested review also. As you only updated the */

**Comment by nandakishormpai** [comment]
Sorry, I didn't notice the comment last time. I have made changes you suggested

**Review by ankitaggarwal23 — APPROVED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**